File: summaryranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if (nums.empty()) return vector<string>{};
        vector<string> res;
        int left = 0, right = 0;
        for (int i = 0; i < nums.size()-1; i++) {
            if (nums[i+1] - nums[i] == 1) right = i+1;
            else {
                if (left == right) res.push_back(to_string(nums[left]));
                else res.push_back(to_string(nums[left]) + "->" + to_string(nums[right]));
                left = i+1;
                right = i+1;
            }
        }
        if (left == right) res.push_back(to_string(nums[left]));
        else res.push_back(to_string(nums[left]) + "->" + to_string(nums[right]));
        return res;
    }
};

File: python/summaryranges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        res = []
        start, end = 0, 0
        if not nums:
            return res
        while end < len(nums):
            
            while end+1 < len(nums) and nums[end+1] == nums[end]+1:
                end += 1
            #print start, end
            if start != end:
                res.append(str(nums[start])+"->"+str(nums[end]))
            else:
                res.append(str(nums[start]))
            start = end = end+1
            
        return res
